### PR TITLE
RavenDB-7592: Refactor new cli + Piped Connectio

### PR DESCRIPTION
**Pull request by aviviadi**

### CODE REVIEW

File: src/Raven.Server/RavenServer.cs [hunk 1]
```diff
@@ -203,49 +203,27 @@ public void Initialize()
...
-                            if (tokens.Length < 1)
+                            var cli = new RavenCli();
+                            var restart = cli.Start(this, writer, reader, false);
+                            if (restart)
                             {
-                                var reply = "Unknown command";
-                                PipeLogAndReply(writer, reply);
-                                continue;
+                                writer.WriteLine("Restarting Server...<DELIMETER_RESTART>");
+                                // TODO :: ADIADI :: perform restart server
```
Comment (ayende) [inline src/Raven.Server/RavenServer.cs:211]: TODO here?

File: src/Raven.Server/RavenServer.cs [hunk 2]
```diff
@@ -203,49 +203,27 @@ public void Initialize()
...
-                                case "init":
-                                    var (apiKey, pubKey) = await ServerStore.GetApiKeyAndPublicKey();
-                                    PipeLogAndReply(writer, $"ApiKey: {apiKey}{Environment.NewLine}PublicKe: {pubKey}");
-                                    writer.Flush();
-                                    break;
-                                default:
-                                    PipeLogAndReply(writer, $"Provided command {tokens[0]} isn't supported");
-                                    continue;
+                                writer.WriteLine("Restarting Server...<DELIMETER_QUIT>");
+                                // TODO :: ADIADI :: perform quit
```
Comment (ayende) [inline src/Raven.Server/RavenServer.cs:216]: TODO here?

File: src/Raven.Server/Utils/RavenCli.cs [hunk 3]
```diff
@@ -67,90 +69,225 @@ internal class RavenCli
...
+            Stats,
+            Info,
+            Gc,
+            LowMem,
+            Help,
+            Logo,
+            Experimental,
+            ImportDir,
+            CreateDb,
+            Logout,
```
Comment (ayende) [inline src/Raven.Server/Utils/RavenCli.cs:98]: What does this mean?

File: src/Raven.Server/RavenServer.cs (hunk 1 above)
Comment (aviviadi) [reply src/Raven.Server/RavenServer.cs:211]: here we identify the remote cli "wants" quit or restart.  The problem is we cannot perform a restart to the Console.ReadLine thread. However we can change few things in order to achieve that differently so it is under "TODO" (if you type in remote connection "q" - you reach here).

File: src/Raven.Server/Utils/RavenCli.cs (hunk 3 above)
Comment (aviviadi) [reply src/Raven.Server/Utils/RavenCli.cs:98]: logout - close gracefully remote cli connection

importdir + createdb : are bypassing REST so they aren't listed in help and cannot be used unless "Experimental on" command is explicitly entered.

**Pull request merged**